src/src_V2/modules/manager/financial/components/payables/Payables.jsx

```diff
@@ -1,302 +1,55 @@
-          <Table aria-labelledby="tableTitle" padding="none">
-            <EnhancedTableHead
-              numSelected={selected.length}
-              order={order}
-              orderBy={orderBy}
-              onSelectAllClick={handleSelectAllClick}
-              onRequestSort={handleRequestSort}
-              rowCount={payables.length}
-            />
-            <TableBody className={styles.tableBody}>
-              {payables
-                .slice()
-                .sort(getComparator(order, orderBy))
-                .slice(page * rowsPerPage, page * rowsPerPage + rowsPerPage)
-                .map((row, index) => {
-                  const isItemSelected = isSelected(row.name);
-                  const labelId = `enhanced-table-checkbox-${index}`;
+    return () => {
+      clearTimeout(timeout);
+    };
+  }
 
-                  return (
-                    <TableRow
-                      hover
-                      onClick={(event) => handleClick(event, row.paymentId)}
-                      role="checkbox"
-                      aria-checked={isItemSelected}
-                      tabIndex={-1}
-                      key={row.name}
-                      selected={isItemSelected}
-                      className={styles.tableRow}
-                    >
-                      <TableCell padding="checkbox" className={styles.tableCheckboxCell}>
-                        <Checkbox
-                          color="primary"
-                          checked={isItemSelected}
-                          inputProps={{
-                            'aria-labelledby': labelId,
-                          }}
-                        />
-                      </TableCell>
-                      {Object.keys(row).map((value) => (
-                        <TableCell className={styles.tableCell} key={row.paymentId}>
-                          <TooltipText text={row[value]} />
-                        </TableCell>
-                      ))}
-                    </TableRow>
-                  );
-                })}
-              {emptyRows > 0 && (
-                <TableRow
-                  style={{
-                    height: 53 * emptyRows,
-                  }}
-                >
-                  <TableCell colSpan={6} />
-                </TableRow>
-              )}
-            </TableBody>
-          </Table>
-          {!payables.length && <NoData />}
-        </TableContainer>
-        {!!payables.length && (
-          <TablePagination
-            rowsPerPageOptions={rowsPerPageOptions}
-            component="div"
-            count={payables.length}
-            rowsPerPage={rowsPerPage}
-            page={page}
-            onPageChange={handleChangePage}
-            onRowsPerPageChange={handleChangeRowsPerPage}
-          />
-        )}
-      </Paper>
-    </Box>
-  );
-};
+  function processPayables(payables) {
+    const processedPayables = [...payables].map((payable) => {
```

Comment:
why are you using rest operator there? It will not help because you are accessing to array's item object nested property. The rest operator doesn't make a deep copy.